File: Project/Backend/src/ArtAuction.Application/Entities/Artist.cs
using ArtAuction.Application.Entities;

namespace ArtAuction.Application.Entities;

public class Artist : ApplicationUser
{
    // Attributes
    public string City { get; set; } 
    public string Country { get; set; }
    public DateTime HireDate { get; set; }
    public int? AdminId { get; set; }
    
    // Relationships
    public Admin Admin { get; set; }
    public ICollection<ArtworkPost> ArtworkPosts { get; set; }
}

File: AuthenticationServices/src/main/java/com/artauction/authenticationservices/Entities/Artist.java
package com.artauction.authenticationservices.Entities;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;
import org.hibernate.annotations.OnDelete;
import org.hibernate.annotations.OnDeleteAction;

import java.time.LocalDateTime;

@Entity
@DiscriminatorValue("Artist")
@Setter
@Getter
public class Artist extends User {
    // Attributes
    @Column(length = 50)
    private String city;

    @Column(length = 50)
    private String country;

    @Column(length = 13)
    private String phoneNumber;
    private LocalDateTime hireDate;

    // Relationships
    @ManyToOne(fetch = FetchType.EAGER)
    @JoinColumn(name = "admin_id")
    @OnDelete(action = OnDeleteAction.NO_ACTION)
    private Admin admin;
}